editoast/src/views/rolling_stocks/mod.rs

```diff
@@ -234,25 +207,35 @@ async fn create(
 async fn update(
     db_pool: Data<DbPool>,
     path: Path<i64>,
-    data: Json<RollingStockForm>,
+    Json(rolling_stock_form): Json<RollingStockForm>,
 ) -> Result<Json<RollingStockWithLiveries>> {
-    let data = data.into_inner();
+    rolling_stock_form.validate()?;
+    let mut db_conn = db_pool.get().await?;
+
     let rolling_stock_id = path.into_inner();
-    let existing_rolling_stock =
-        retrieve_existing_rolling_stock(&db_pool, rolling_stock_id).await?;
-    assert_rolling_stock_unlocked(existing_rolling_stock.clone())?;
-    let version = existing_rolling_stock.version;
-
-    let mut rolling_stock = data.into_rolling_stock_model(rolling_stock_id, version);
-    rolling_stock.validate()?;
-    if existing_rolling_stock != rolling_stock {
-        rolling_stock.version = version.map(|v| v + 1);
+
+    let existing_rs = retrieve_existing_rolling_stock(&db_pool, rolling_stock_id).await?;
```

Comment:
I'd suggest to `retrieve` directly here, this function is useful when you don't have a connection open, but here we do.